src/olympia/addons/views.py

```diff
@@ -346,20 +346,19 @@ def get_georestrictions(self):
 
     @action(
         detail=True,
-        methods=['get', 'patch'],
         serializer_class=AddonEulaPolicySerializer,
         # For this action, developers use the same serializer - it only
         # contains eula/privacy policy.
         serializer_class_for_developers=AddonEulaPolicySerializer,
     )
     def eula_policy(self, request, pk=None):
-        kwargs = {}
-        if request.method in SAFE_METHODS:
-            method = self.retrieve
-        else:
-            kwargs['partial'] = True
-            method = self.update
-        return method(request, **kwargs)
+        return self.retrieve(request)
+
+    @eula_policy.mapping.patch
+    def update_eula_policy(self, request, pk=None):
+        kwargs = {'partial': True}
+        self.permission_classes = self.write_permission_classes
+        return self.update(request, **kwargs)
```

Comment:
nit, the kwargs dict seems unnecessary?
```suggestion
        self.permission_classes = self.write_permission_classes
        return self.update(request, partial=True)
```